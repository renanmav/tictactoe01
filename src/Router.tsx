import React from 'react';
import { createStackNavigator } from '@react-navigation/stack';
import { NavigationContainer } from '@react-navigation/native';

import Menu from './screens/Menu';
import SelectDifficulty, { Difficulties } from './screens/SelectDifficulty';
import Game from './screens/Game';

export type MainStackParamList = {
  Menu: undefined;
  SelectDifficulty: undefined;
  Game: { difficulty: Difficulties };
};

const MainStack = createStackNavigator<MainStackParamList>();

const Router: React.FC = () => {
  return (
    <NavigationContainer>
      <MainStack.Navigator
        initialRouteName="Menu"
        headerMode="none"
        screenOptions={{
          animationEnabled: false,
        }}>
        <MainStack.Screen name="Menu" component={Menu} />
        <MainStack.Screen name="SelectDifficulty" component={SelectDifficulty} />
        <MainStack.Screen name="Game" component={Game} />
      </MainStack.Navigator>
    </NavigationContainer>
  );
};

export default Router;
