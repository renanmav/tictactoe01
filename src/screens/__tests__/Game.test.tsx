import React from 'react';
import { fireEvent, render, wait } from '@testing-library/react-native';

import Router from '../../Router';

it('should render correctly', async () => {
  const { getByText, getByTestId } = render(<Router />);

  fireEvent.press(getByText('Jogar'));
  fireEvent.press(getByText('FÃ¡cil'));

  await wait(() => {
    expect(getByTestId('game')).toMatchSnapshot();
    expect(getByTestId('game-top-container')).toMatchSnapshot();
    expect(getByTestId('game-content-container')).toMatchSnapshot();
  });
});
