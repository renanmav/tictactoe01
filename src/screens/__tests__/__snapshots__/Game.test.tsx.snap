// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Image
  source={1}
  style={
    Array [
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      Object {
        "height": undefined,
        "width": undefined,
      },
      undefined,
    ]
  }
  testID="game"
/>
`;

exports[`should render correctly 2`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 32,
      },
      Object {
        "width": "100%",
      },
      Object {
        "alignItems": "flex-start",
      },
      Object {
        "marginVertical": 24,
      },
    ]
  }
  testID="game-top-container"
>
  <TouchableOpacity
    activeOpacity={0.2}
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "resizeMode": "contain",
          },
          Object {
            "height": 35,
            "width": 35,
          },
        ]
      }
    />
  </TouchableOpacity>
</View>
`;

exports[`should render correctly 3`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 32,
      },
      Object {
        "flex": 1,
      },
      Object {
        "justifyContent": "center",
      },
      Object {
        "alignItems": "center",
      },
    ]
  }
  testID="game-content-container"
>
  <View
    style={
      Array [
        Object {
          "width": "100%",
        },
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "width": "25%",
          },
          Object {
            "alignItems": "center",
          },
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "resizeMode": "contain",
            },
            Object {
              "marginBottom": 20,
            },
            Object {
              "height": 50,
              "width": 50,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
            },
            Object {
              "fontSize": 20,
            },
            Object {
              "fontWeight": "bold",
            },
          ]
        }
      >
        VocÃª
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flexGrow": 1,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "width": "25%",
          },
          Object {
            "alignItems": "center",
          },
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "resizeMode": "contain",
            },
            Object {
              "marginBottom": 20,
            },
            Object {
              "height": 50,
              "width": 50,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
            },
            Object {
              "fontSize": 20,
            },
            Object {
              "fontWeight": "bold",
            },
          ]
        }
      >
        Bot
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginBottom": 50,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#2D3748",
        },
        Object {
          "width": "100%",
        },
        Object {
          "borderRadius": 4,
        },
        Object {
          "flexDirection": "row",
        },
        Object {
          "flexWrap": "wrap",
        },
        Object {
          "padding": 4,
        },
        Object {
          "aspectRatio": 1,
        },
      ]
    }
    testID="tictactoe-board"
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
          },
          Object {
            "width": "33.333333%",
          },
          Object {
            "padding": 4,
          },
          Object {
            "aspectRatio": 1,
          },
        ]
      }
    >
      <TouchableWithoutFeedback
        testID="field-touchable"
      >
        <View
          style={
            Array [
              Array [
                Object {
                  "backgroundColor": "#242c3a",
                },
                Object {
                  "borderRadius": 4,
                },
                Object {
                  "flex": 1,
                },
                Object {
                  "alignItems": "center",
                },
                Object {
                  "justifyContent": "center",
                },
                Object {
                  "overflow": "hidden",
                },
              ],
              false,
            ]
          }
          testID="field-background"
        />
      </TouchableWithoutFeedback>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
          },
          Object {
            "width": "33.333333%",
          },
          Object {
            "padding": 4,
          },
          Object {
            "aspectRatio": 1,
          },
        ]
      }
    >
      <TouchableWithoutFeedback
        testID="field-touchable"
      >
        <View
          style={
            Array [
              Array [
                Object {
                  "backgroundColor": "#242c3a",
                },
                Object {
                  "borderRadius": 4,
                },
                Object {
                  "flex": 1,
                },
                Object {
                  "alignItems": "center",
                },
                Object {
                  "justifyContent": "center",
                },
                Object {
                  "overflow": "hidden",
                },
              ],
              false,
            ]
          }
          testID="field-background"
        />
      </TouchableWithoutFeedback>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
          },
          Object {
            "width": "33.333333%",
          },
          Object {
            "padding": 4,
          },
          Object {
            "aspectRatio": 1,
          },
        ]
      }
    >
      <TouchableWithoutFeedback
        testID="field-touchable"
      >
        <View
          style={
            Array [
              Array [
                Object {
                  "backgroundColor": "#242c3a",
                },
                Object {
                  "borderRadius": 4,
                },
                Object {
                  "flex": 1,
                },
                Object {
                  "alignItems": "center",
                },
                Object {
                  "justifyContent": "center",
                },
                Object {
                  "overflow": "hidden",
                },
              ],
              false,
            ]
          }
          testID="field-background"
        />
      </TouchableWithoutFeedback>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
          },
          Object {
            "width": "33.333333%",
          },
          Object {
            "padding": 4,
          },
          Object {
            "aspectRatio": 1,
          },
        ]
      }
    >
      <TouchableWithoutFeedback
        testID="field-touchable"
      >
        <View
          style={
            Array [
              Array [
                Object {
                  "backgroundColor": "#242c3a",
                },
                Object {
                  "borderRadius": 4,
                },
                Object {
                  "flex": 1,
                },
                Object {
                  "alignItems": "center",
                },
                Object {
                  "justifyContent": "center",
                },
                Object {
                  "overflow": "hidden",
                },
              ],
              false,
            ]
          }
          testID="field-background"
        />
      </TouchableWithoutFeedback>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
          },
          Object {
            "width": "33.333333%",
          },
          Object {
            "padding": 4,
          },
          Object {
            "aspectRatio": 1,
          },
        ]
      }
    >
      <TouchableWithoutFeedback
        testID="field-touchable"
      >
        <View
          style={
            Array [
              Array [
                Object {
                  "backgroundColor": "#242c3a",
                },
                Object {
                  "borderRadius": 4,
                },
                Object {
                  "flex": 1,
                },
                Object {
                  "alignItems": "center",
                },
                Object {
                  "justifyContent": "center",
                },
                Object {
                  "overflow": "hidden",
                },
              ],
              false,
            ]
          }
          testID="field-background"
        />
      </TouchableWithoutFeedback>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
          },
          Object {
            "width": "33.333333%",
          },
          Object {
            "padding": 4,
          },
          Object {
            "aspectRatio": 1,
          },
        ]
      }
    >
      <TouchableWithoutFeedback
        testID="field-touchable"
      >
        <View
          style={
            Array [
              Array [
                Object {
                  "backgroundColor": "#242c3a",
                },
                Object {
                  "borderRadius": 4,
                },
                Object {
                  "flex": 1,
                },
                Object {
                  "alignItems": "center",
                },
                Object {
                  "justifyContent": "center",
                },
                Object {
                  "overflow": "hidden",
                },
              ],
              false,
            ]
          }
          testID="field-background"
        />
      </TouchableWithoutFeedback>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
          },
          Object {
            "width": "33.333333%",
          },
          Object {
            "padding": 4,
          },
          Object {
            "aspectRatio": 1,
          },
        ]
      }
    >
      <TouchableWithoutFeedback
        testID="field-touchable"
      >
        <View
          style={
            Array [
              Array [
                Object {
                  "backgroundColor": "#242c3a",
                },
                Object {
                  "borderRadius": 4,
                },
                Object {
                  "flex": 1,
                },
                Object {
                  "alignItems": "center",
                },
                Object {
                  "justifyContent": "center",
                },
                Object {
                  "overflow": "hidden",
                },
              ],
              false,
            ]
          }
          testID="field-background"
        />
      </TouchableWithoutFeedback>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
          },
          Object {
            "width": "33.333333%",
          },
          Object {
            "padding": 4,
          },
          Object {
            "aspectRatio": 1,
          },
        ]
      }
    >
      <TouchableWithoutFeedback
        testID="field-touchable"
      >
        <View
          style={
            Array [
              Array [
                Object {
                  "backgroundColor": "#242c3a",
                },
                Object {
                  "borderRadius": 4,
                },
                Object {
                  "flex": 1,
                },
                Object {
                  "alignItems": "center",
                },
                Object {
                  "justifyContent": "center",
                },
                Object {
                  "overflow": "hidden",
                },
              ],
              false,
            ]
          }
          testID="field-background"
        />
      </TouchableWithoutFeedback>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
          },
          Object {
            "width": "33.333333%",
          },
          Object {
            "padding": 4,
          },
          Object {
            "aspectRatio": 1,
          },
        ]
      }
    >
      <TouchableWithoutFeedback
        testID="field-touchable"
      >
        <View
          style={
            Array [
              Array [
                Object {
                  "backgroundColor": "#242c3a",
                },
                Object {
                  "borderRadius": 4,
                },
                Object {
                  "flex": 1,
                },
                Object {
                  "alignItems": "center",
                },
                Object {
                  "justifyContent": "center",
                },
                Object {
                  "overflow": "hidden",
                },
              ],
              false,
            ]
          }
          testID="field-background"
        />
      </TouchableWithoutFeedback>
    </View>
  </View>
</View>
`;
